contracts/thecosomata/ThecosomataETH.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IBTRFLY is IERC20 {
+    function burn(uint256 amount) external;
+
+    function decimals() external view returns (uint8);
+}
+
+interface IRedactedTreasury {
+    function manage(address _token, uint256 _amount) external;
+}
+
+interface ICurveCryptoPool {
+    function add_liquidity(uint256[2] calldata amounts, uint256 min_mint_amount)
+        external
+        payable;
+
+    function calc_token_amount(uint256[2] calldata amounts)
+        external
+        view
+        returns (uint256);
+
+    // Would be replaced by Chainlink based oracle
+    function price_oracle() external view returns (uint256);
+
+    function token() external view returns (address);
+}
+
+contract ThecosomataETH is Ownable {
```

Comment:
A Code4rena comment made me realize that we should use the AccessControl instead of Ownable, because we should have a new role that is allowed to call `perofrmUpkeep` but not `withdraw`

Reply:
Explaining the above: we want to do this because the owner/admin should be the multisig and should be the only one able to withdraw tokens. We should limit our own off-chain process or keeper contracts to calling only `performUpkeep`

Reply:
That makes sense!